### PR TITLE
GROOVY-8917: Failed to infer parameter type of some SAM, e.g. BinaryO…

### PR DESCRIPTION
**Pull request by danielsun1106**
…perator

### CODE REVIEW

File: src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java [hunk 1]
```diff
@@ -213,4 +220,15 @@ private static String makeRefDescription(String name) {
...
+    private static final Map<ClassNode, ClassNode> PRIMITIVE_CLASSNODE_TO_WRAPPED_CLASSNODE_MAP = Maps.of(
+            ClassHelper.byte_TYPE, ClassHelper.Byte_TYPE,
+            ClassHelper.boolean_TYPE, ClassHelper.Boolean_TYPE,
+            ClassHelper.char_TYPE, ClassHelper.Character_TYPE,
+            ClassHelper.double_TYPE, ClassHelper.Double_TYPE,
+            ClassHelper.float_TYPE, ClassHelper.Float_TYPE,
+            ClassHelper.int_TYPE, ClassHelper.Integer_TYPE,
+            ClassHelper.long_TYPE, ClassHelper.Long_TYPE,
+            ClassHelper.short_TYPE, ClassHelper.Short_TYPE
+    );
```
Comment (paulk-asert) [inline src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java]: Isn't this already in ClassHelper#PRIMITIVE_TYPE_TO_WRAPPER_TYPE_MAP?

Comment (danielsun1106) [reply src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java]: @paulk-asert As we can see, the existing type mapping is Class to Class,  but we need map ClassNode to ClassNode. Though we can convert ClassNode to Class and get the mapping result then convert the result to ClassNode, it is less efficient.

https://github.com/apache/groovy/blob/9cea59b95367c1d563e719ebeddcdecb58199750/src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java#L213

Comment (paulk-asert) [reply src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java]: Both appear to be Map<ClassNode, ClassNode> to me?

Comment (paulk-asert) [reply src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java]: I was referring to `ClassHelper#PRIMITIVE_TYPE_TO_WRAPPER_TYPE_MAP` not `TypeUtil#PRIMITIVE_TYPE_TO_WRAPPED_CLASS_MAP`.

Comment (paulk-asert) [reply src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java]: And I guess also `TypeUtil#PRIMITIVE_TYPE_TO_WRAPPED_CLASS_MAP` would be better named `PRIMITIVE_CLASS_TO_WRAPPED_CLASS_MAP`.

Comment (danielsun1106) [reply src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java]: Gotcha. I will use the map you suggest instead.

Comment (danielsun1106) [reply src/main/java/org/codehaus/groovy/classgen/asm/util/TypeUtil.java]: @paulk-asert I've refined the PR according to your suggestion ;-)

**Pull request closed without merging**